EmployeeController.java

```diff
@@ -29,38 +35,43 @@ public class EmployeeController {
     private EmployeeRepository employeeData;
 
     @RequestMapping(value = "/addNewEmployee", method = RequestMethod.POST)
-    public Employee addEmployee(@RequestBody Employee employee) {
+    public ResponseEntity<Object> addEmployee(@RequestBody Employee employee) {
 //get data from each variable and do validations
         EmployeeValidator validator = new EmployeeValidator();
-        if(validator.isPhoneNumberValid(employee.getPhoneNumber()))
-        return employeeData.save(employee);
+        if(validator.isPhoneNumberValid(employee.getPhoneNumber()) && validator.isDojValid(employee.getDateOfJoining()) && validator.isEmailValid(employee.getEmail())) {
+            employeeData.save(employee);
+            return new ResponseEntity<Object>("Employee successfully added in DB",HttpStatus.OK);
+        }
         else
-            throw new InvalidEmployeeDataException("invalid phone number entered");
-
+        return new ResponseEntity<Object>(new InvalidEmployeeDataException("invalid data entered"), HttpStatus.OK);
     }
 
     @RequestMapping(value = "/listEmployees", method = RequestMethod.GET)
-    public List<EmployeeTax> getEmployeesTaxInfo() {
-        List<Employee> allEmployees = (List<Employee>) employeeData.findAll().iterator();
+    public ResponseEntity<Object> getEmployeesTaxInfo() {
+        List<Employee> allEmployees = ((ArrayList<Employee>) employeeData.findAll());
         List<EmployeeTax> taxedEmployees = new ArrayList<>();
 
         for (Employee employee:allEmployees) {
             EmployeeTax taxData = null;
             taxData = new EmployeeTax();
-            long salaryForTax=findSalaryForTax(employee.getSalary(),Integer.parseInt(employee.getDateOfJoining().substring(5,6)),Integer.parseInt(employee.getDateOfJoining().substring(8,9)));
+            List<String> date=Collections.list(new StringTokenizer(employee.getDateOfJoining(),"-")).stream().map(token->(String)token).toList();

```

Comment:
Could have converted to Date instead using strings

Reply:
Agreed. In validator I have used SimpleDateFormat to verify the date. So from date object I can get day, month, year, which is much easier. I am aware of this, but purposefully I used string manipulations to let you know I am aware of these too, which are indeed needed in many cases while developing java applications. As I already checking with formatter, I am sure I always get proper string date here, so I tried this here. Hope this clarifies. Thanks